File: src/test/java/edu/kit/iti/formal/stvs/view/menu/StvsMenuBarDemo.java
package edu.kit.iti.formal.stvs.view.menu;

import edu.kit.iti.formal.stvs.model.StvsRootModel;
import edu.kit.iti.formal.stvs.Demo;
import edu.kit.iti.formal.stvs.view.JavaFxTest;
import javafx.application.Application;
import javafx.beans.property.SimpleObjectProperty;
import javafx.scene.Scene;
import javafx.scene.layout.VBox;
import org.junit.Test;
import org.junit.experimental.categories.Category;

/**
 * Created by csicar on 30.01.17.
 */
@Category(Demo.class)
public class StvsMenuBarDemo {

  @Test
  public void javaFxTest() {
    JavaFxTest.setToBeViewed(this::simpleScene);
    Application.launch(JavaFxTest.class);
  }

  private Scene simpleScene() {
    Scene scene = new Scene(new VBox(), 400, 350);

    StvsRootModel rootModel = new StvsRootModel();

    StvsMenuBarController menuBarController = new StvsMenuBarController(new
        SimpleObjectProperty<>(rootModel));
    ((VBox) scene.getRoot()).getChildren().addAll(menuBarController.getView());

    return scene;
  }
}

File: stvs/src/test/kotlin/edu/kit/iti/formal/stvs/view/menu/StvsMenuBarDemo.kt
package edu.kit.iti.formal.stvs.view.menu

import org.junit.jupiter.api.Tag
import edu.kit.iti.formal.stvs.model.StvsRootModel
import edu.kit.iti.formal.stvs.view.JavaFxTest
import javafx.application.Application
import javafx.beans.property.SimpleObjectProperty
import javafx.scene.Scene
import javafx.scene.layout.VBox
import org.junit.jupiter.api.Test
import java.util.function.Supplier

/**
 * Created by csicar on 30.01.17.
 */
@Tag("demo")
class StvsMenuBarDemo {
    @Test
    fun javaFxTest() {
        JavaFxTest.Companion.setToBeViewed(Supplier<Scene?> { this.simpleScene() })
        Application.launch(JavaFxTest::class.java)
    }

    private fun simpleScene(): Scene {
        val scene = Scene(VBox(), 400.0, 350.0)

        val rootModel = StvsRootModel()

        val menuBarController = StvsMenuBarController(SimpleObjectProperty(rootModel))
        (scene.root as VBox).children.addAll(menuBarController.view)

        return scene
    }
}
